railties/lib/rails/railtie.rb

```diff
@@ -135,6 +136,7 @@ module Rails
   class Railtie
     autoload :Configuration, "rails/railtie/configuration"
 
+    extend ActiveSupport::DescendantsTracker
```

Comment:
Why this change? This is adding a lot of methods to railties that we don't need to add.

Reply:
This is to prevent `Railtie.subclasses` from being overridden by `DescendantsTracker#subclasses` when `ActiveSupport::Callbacks` is included in `Engine` (because `ActiveSupport::Callbacks` also [extends `DescendantsTracker`](https://github.com/rails/rails/blob/cf4fa5caf39f1271d3f8c4c9774e874430ea0f68/activesupport/lib/active_support/callbacks.rb#L66)).  If `Railtie.subclasses` is overridden, then it will include abstract railties, which will raise an error when [instantiated via `Railties#initialize`](https://github.com/rails/rails/blob/cf4fa5caf39f1271d3f8c4c9774e874430ea0f68/railties/lib/rails/engine/railties.rb#L10-L11).

An alternative would be to move the abstract railties filtering logic to `Railties#initialize`, though that would change the contract of `Railtie.subclasses`.

It's worth mentioning, though, that `DescendantsTracker` would only add `::descendants` and `::direct_descendants` methods to railties.